uefi/src/table/runtime.rs

```diff
@@ -582,6 +582,68 @@ impl Display for Time {
     }
 }
 
+impl TryFrom<&[u8]> for Time {
+    type Error = TimeError;
+
+    fn try_from(bytes: &[u8]) -> core::result::Result<Self, Self::Error> {
+        if size_of::<Time>() <= bytes.len() {
+            let (year, bytes) = bytes.split_at(size_of::<u16>());
+            // The byte slice is at least as large as [`Time`], so this cannot panic.
+            let year = u16::from_le_bytes(year.try_into().unwrap());
+
+            let (month, bytes) = bytes.split_at(size_of::<u8>());
+            let month = u8::from_le_bytes(month.try_into().unwrap());
+
+            let (day, bytes) = bytes.split_at(size_of::<u8>());
+            let day = u8::from_le_bytes(day.try_into().unwrap());
+
+            let (hour, bytes) = bytes.split_at(size_of::<u8>());
+            let hour = u8::from_le_bytes(hour.try_into().unwrap());
+
+            let (minute, bytes) = bytes.split_at(size_of::<u8>());
+            let minute = u8::from_le_bytes(minute.try_into().unwrap());
+
+            let (second, bytes) = bytes.split_at(size_of::<u8>());
+            let second = u8::from_le_bytes(second.try_into().unwrap());
+
+            // Account for padding
+            let (_, bytes) = bytes.split_at(size_of::<u8>());
+
+            let (nanosecond, bytes) = bytes.split_at(size_of::<u32>());
+            let nanosecond = u32::from_le_bytes(nanosecond.try_into().unwrap());
+
+            let (tz, bytes) = bytes.split_at(size_of::<i16>());
+            let time_zone: Option<i16> = match i16::from_le_bytes(tz.try_into().unwrap()) {
+                Self::UNSPECIFIED_TIMEZONE => None,
+                num => Some(num),
+            };
+
+            let (daylight, _) = bytes.split_at(size_of::<u8>());
+            let daylight = u8::from_le_bytes(daylight.try_into().unwrap());
+            let daylight = Daylight::from_bits(daylight).ok_or(TimeError {
+                daylight: true,
+                ..Default::default()
+            })?;
+
+            let time_params = TimeParams {
+                year,
+                month,
+                day,
+                hour,
+                minute,
+                second,
+                nanosecond,
+                time_zone,
+                daylight,
+            };
+
+            Time::new(time_params)
+        } else {
+            Err(TimeError::default())
```

Comment:
I wasn't sure what to do here: adding a `byteconversion` field to TimeError didn't make sense. Perhaps I can modify `Display` so that it treats a `TimeError::default()` as an error pertaining to byte conversion, and writes a string accordingly? Or I just don't upstream this.

Reply:
I think make a new error type for this. `enum TimeByteConversionError { NotEnoughData, OutOfRange(TimeError) }`, something like that.